### PR TITLE
Add GitHub Action to use Carpenter CI

### PR DESCRIPTION
**Pull request by mfleader**
## Changes introduced with this PR

Add a GitHub action to call Carpenter CI to check this project meets our plugin requirements, build the plugin image, and push it to a chosen registry.

Converge project towards Arcaflow Python Plugin requirements.
---
By contributing to this repository, I agree to the [contribution guidelines](https://github.com/arcalot/.github/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: .github/workflows/carpenter.yaml [hunk 1]
```diff
@@ -5,27 +5,20 @@ on:
...
-      # #   run: docker pull ghcr.io/mfleader/arcaflow-plugin-image-builder:gh-container-axn
-      # - name: carpenter build
-      #   # run: docker run --rm -v /var/run/docker.sock:/var/run/docker.sock:z -v ${{ github.workspace }}:/github/workspace ghcr.io/mfleader/arcaflow-plugin-image-builder:gh-container-axn /carpenter build >> $GITHUB_STEP_SUMMARY
-      #   uses: mfleader/arcaflow-plugin-image-builder@gh-container-axn
-      #   with:
-      #     args: build --build --push
+      - name: Checkout this project
+        uses: actions/checkout@v3
+      - name: carpenter build
+        uses: mfleader/arcaflow-plugin-image-builder@gh-container-axn
```
[the remark this answers is not in the document]
Comment (sandrobonazzola) [reply .github/workflows/carpenter.yaml:20]: shouldn't we get the container built within arcalot org? @janosdebugs ?

[the remark this answers is not in the document]
Comment (mfleader) [reply .github/workflows/carpenter.yaml:20]: @sandrobonazzola what do you mean by 'built within arcalot org'?

File: example_plugin.py [hunk 2]
```diff
@@ -27,33 +30,39 @@ class ErrorOutput:
...
     This is the output data structure in the error  case.
     """
+
     error: str
 
 
-# The following is a decorator (starting with @). We add this in front of our function to define the metadata for our
-# step.
+# The following is a decorator (starting with @). We add this in front of our
+# function to define the metadata for our step.
```
Comment (portante) [inline example_plugin.py:38]: Perhaps consider:

```
# We use the `plugin.step` decorator in front of our function to provide the
# definition of the metadata for our step.
```

File: .github/workflows/build.yaml [hunk 3]
```diff
@@ -30,3 +30,4 @@ jobs:
           name: coverage
           path: htmlcov
           if-no-files-found: error
+
```
Comment (portante) [inline .github/workflows/build.yaml:33]: It seems that some files are changed with additional new lines, and others have no new lines.  Is this a feature of some IDE?

**Review by portante — COMMENTED** [review]
Just a few comments to consider.

**Review by portante — COMMENTED** [review]
Looks good.

**Review by sandrobonazzola — APPROVED** [review]
LGTM

**Pull request merged**